test/test_type_promotion.py

```diff
@@ -442,26 +442,57 @@ def test_transpose(self, device):
     @dtypes(torch.bool, torch.short, torch.uint8, torch.int, torch.long)
     @float_double_default_dtype
     def test_true_divide(self, device, dtype):
-        dividend = torch.randn(5, device=device).to(dtype)
+        dividend = (torch.randn(5, device=device) * 100).to(dtype)
         divisor = torch.arange(1, 6, device=device).to(dtype)
+
+        # Tests tensor/tensor division
         casting_result = dividend.to(torch.get_default_dtype()) / divisor.to(torch.get_default_dtype())
         self.assertEqual(casting_result, torch.true_divide(dividend, divisor))
 
+        # Tests tensor/scalar division
+        casting_result = dividend.to(torch.get_default_dtype()) / 2
+        self.assertEqual(casting_result, torch.true_divide(dividend, 2.))
+
     @onlyOnCPUAndCUDA
-    @dtypes(torch.bool, torch.short, torch.uint8, torch.int, torch.long)
+    @dtypes(torch.float, torch.double,
+            torch.bool, torch.short, torch.uint8, torch.int, torch.long)
     def test_true_divide_out(self, device, dtype):
-        dividend = torch.randn(5, device=device).to(dtype)
+        dividend = (torch.randn(5, device=device) * 100).to(dtype)
         divisor = torch.arange(1, 6, device=device).to(dtype)
 
         # Tests that requests for an integer quotient fail
-        integral_quotient = torch.empty(5, device=device, dtype=dtype)
-        with self.assertRaises(RuntimeError):
-            torch.true_divide(dividend, divisor, out=integral_quotient)
+        if dtype in (torch.bool, torch.short, torch.uint8, torch.int, torch.long):
```

Comment:
super nit: it's slightly nicer if you do all bit-length variants, easier to read (since you have to for uint8).

Reply:
Super nits deserve love, too.

Reply:
Fixed.